mypy/semanal.py

```diff
@@ -2210,8 +2210,8 @@ def check_no_global(self, n: str, ctx: Context,
                         is_func: bool = False) -> None:
         if n in self.globals:
             if is_func and isinstance(self.globals[n].node, FuncDef):
-                self.fail(("Name '{}' already defined (overload variants "
-                           "must be next to each other)").format(n), ctx)
+                self.fail(("Name '{}' already defined on line "
+                           "{}").format(n, self.globals[n].node.line), ctx)
```

Comment:
Occasionally a node has no line number, and then `line` is `-1`. Can you check for that?

You could also consider emitting the line number of the previous definition as a separate message with `severity='note'`. That makes it possible for e.g. Emacs to jump to that line separately.